### PR TITLE
GRO-1846: map offers to middleware create application event

### PR DESCRIPTION
**Pull request by KastorDriver**
#### Description
[GRO-1846](https://selina.atlassian.net/browse/GRO-1846)

In addition to the previous [PR-171](https://github.com/Selina-Finance/ms-lending-api/pull/171) (please, check its description for more context) we need to enrich the middleware create application Kafka event with offers that the Lending API gets from the ms-selection service, so the MW could create and save applications without excess requests to the Decisioning service.

#### Background Context
At stage 5 we need to create a Middleware create application Kafka event. This event should contain not only request params (that we get at stage 1) but a list of offers (gotten at stage 4).

The Kafka [event model](https://www.notion.so/Kafka-Model-08f4585a453f453bb7649828b83260a9) has been updated with information about the expected offers list.

#### Additional Information
image

[GRO-1846]: https://selina.atlassian.net/browse/GRO-1846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kastordriver**
You can assign the PR to them by writing `/assign @kastordriver` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **jx-selina-finance-ms-lending-api-pr-173** [here](http://ms-lending-api-pr173.selinadev.co.uk)

File: src/main/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapper.java [hunk 1]
```diff
@@ -24,17 +28,60 @@ public abstract class MiddlewareCreateApplicationEventMapper {
...
+                                                                                           List<Product> products);
 
     Fees createDefaultFees() {
         return Fees.builder()
                 .isAddProductFeesToFacility(ADD_PRODUCT_FEES_TO_FACILITY)
                 .build();
     }
+
+    @Mapping(target = "id", source = "product.offer.id")
+    @Mapping(target = "productFeeCanAdd", source = "product.offer.canAddProductFee")
```
Comment (WeiLu1) [inline src/main/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapper.java:49]: I see you are mapping the two names here (which answers my question above), why can't we keep it the same name by choosing one?

File: src/test/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapperTest.java [hunk 2]
```diff
@@ -6,10 +6,14 @@
...
 import com.selina.lending.internal.service.application.domain.Address;
 import com.selina.lending.internal.service.application.domain.Applicant;
+import com.selina.lending.internal.service.application.domain.Erc;
+import com.selina.lending.internal.service.application.domain.Fees;
 import com.selina.lending.internal.service.application.domain.Incomes;
 import com.selina.lending.internal.service.application.domain.LoanInformation;
+import com.selina.lending.internal.service.application.domain.Offer;
 import com.selina.lending.internal.service.application.domain.PropertyDetails;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
```
Comment (WeiLu1) [inline src/test/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapperTest.java:16]: import not used, can delete

File: src/test/java/com/selina/lending/messaging/publisher/MiddlewareCreateApplicationEventPublisherTest.java [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import com.selina.lending.internal.exception.KafkaSendEventException;
 import com.selina.lending.internal.mapper.MapperBase;
 import com.selina.lending.internal.service.TokenService;
+import com.selina.lending.internal.service.application.domain.quote.Product;
```
Comment (WeiLu1) [inline src/test/java/com/selina/lending/messaging/publisher/MiddlewareCreateApplicationEventPublisherTest.java:6]: unused import, can delete

**Comment by selina-bot-user** [comment]
New changes are detected. LGTM label has been removed.

File: src/test/java/com/selina/lending/messaging/publisher/MiddlewareCreateApplicationEventPublisherTest.java (hunk 3 above)
Comment (KastorDriver) [reply src/test/java/com/selina/lending/messaging/publisher/MiddlewareCreateApplicationEventPublisherTest.java]: Thanks!
Deleted

File: src/test/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapperTest.java (hunk 2 above)
Comment (KastorDriver) [reply src/test/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapperTest.java]: Deleted

File: src/main/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapper.java (hunk 1 above)
Comment (KastorDriver) [reply src/main/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapper.java:49]: Good question!

The "productFeeCanAdd" is the property the MW expects to get in a Kafka event.
The "canAddProductFee" is the one we get from the ms-selection service.

So, these fields hold the same value but just were named differently.

Comment (WeiLu1) [reply src/main/java/com/selina/lending/messaging/mapper/middleware/MiddlewareCreateApplicationEventMapper.java:49]: image

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **jx-selina-finance-ms-lending-api-pr-173** [here](http://ms-lending-api-pr173.selinadev.co.uk)

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **jx-selina-finance-ms-lending-api-pr-173** [here](http://ms-lending-api-pr173.selinadev.co.uk)

**Pull request merged**